src/metron/networkreader/network_reader_test.go

```diff
@@ -8,16 +8,29 @@ import (
 	"metron/writers/mocks"
 
 	"github.com/cloudfoundry/dropsonde/metric_sender/fake"
-	"github.com/cloudfoundry/loggregatorlib/loggertesthelper"
-
 	"github.com/cloudfoundry/dropsonde/metricbatcher"
 	"github.com/cloudfoundry/dropsonde/metrics"
+	"github.com/cloudfoundry/loggregatorlib/loggertesthelper"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
 	"strconv"
 	"time"
 )
 
+func randomPort() int {
+	addr, err := net.ResolveUDPAddr("udp", "1.2.3.4:1")
```

Comment:
Just to clarify what this function is doing. You are not using net.Listen("udp", ":0") because the library doesn't allow it. So you make a UDP connection to a random IP address and port and then you use the local port assigned to the connection as the random port which the test will use. And this works because you are closing the connection before you return the port. Correct?